packages/rehype-shikiji/src/index.ts

```diff
@@ -70,6 +70,18 @@ const rehypeShikiji: Plugin<[RehypeShikijiOptions], Root> = function (options =
       }
 
       const attrs = (head.data as any)?.meta
+
+      if (attrs) {
+        const meta = Object.fromEntries(attrs.split(' ').reduce((prev: [string, boolean | string][], curr: string) => {
+          const [key, value] = curr.split('=')
+          const isNormalKey = /^[A-Za-z0-9]+$/.test(key)
+          if (isNormalKey)
+            prev = [...prev, [key, value || true]]
+          return prev
+        }, []))
+        codeOptions.meta = meta
```

Comment:
I am not so sure about mutating the option. Instead maybe we could have a hook function like `handleMeta` or `onMeta`?

Reply:
![image](https://github.com/antfu/shikiji/assets/45410431/9c6be379-e39c-4901-9067-9f28025bbbcb)

Is that what you mean?